pandas/core/dtypes/cast.py

```diff
@@ -1505,12 +1505,20 @@ def cast_scalar_to_array(shape, value, dtype: Optional[DtypeObj] = None) -> np.n
 
     """
     if dtype is None:
-        dtype, fill_value = infer_dtype_from_scalar(value)
+        dtype, fill_value = infer_dtype_from_scalar(value, pandas_dtype=True)
     else:
         fill_value = value
 
-    values = np.empty(shape, dtype=dtype)
-    values.fill(fill_value)
+    # TODO: Update this function to add support for 3rd party extension types
+    # Issue #34959
+    if is_extension_array_dtype(dtype):
+        if isinstance(shape, int):
+            shape = (shape, 1)
+        value = [construct_1d_arraylike_from_scalar(value, shape[0], dtype)]
```

Comment:
no please
don’t do this here

construct_1s_arraylike can be updated to handle this if needed

Reply:
What's the requested change? We don't want to change `construct_1d_arraylike_from_scalar` to return a list of arrays.

Reply:
@jreback Happy to make whatever changes are needed, but also want to clarify your request.

Currently `construct_1d_arraylike_from_scalar` takes in value, length, and dtype. It then returns a "1d" array similar to `[value] * length`.

Are you suggesting that we change `construct_1d_arraylike_from_scalar` to something like `construct_arraylike_from_scalar` (or maybe `construct_matrixlike_from_scalar`) and allow it to accept a shape (tuple) argument in addition to a length (int) argument?

Reply:
Hopefully https://github.com/pandas-dev/pandas/pull/34875#discussion_r446180430 helps here. Ideally we have

1. One function for casting a scalar to a 1D extension array, which we call several times to create a list of extension ararys
2. One function for casting a scalar to a 2D ndarray (we already have that).